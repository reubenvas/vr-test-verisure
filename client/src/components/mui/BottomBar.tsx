import React from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import scenarioStore from '../../store/scenarioStore';
import { postScenarios } from '../../api/scenarioApi';

const useStyles = makeStyles(() => createStyles({
    appBar: {
        top: 'auto',
        bottom: 0,
    },
    popoutButton: {
        position: 'absolute',
        zIndex: 1,
        top: -15,
        left: 0,
        right: 0,
        margin: '0 0',
    },
}));

const sendScenaroisToServerAndRedirect = (): void => {
    console.log('generating report');
    const scenarios = JSON.stringify(scenarioStore.scenarios);
    postScenarios(scenarios);
    // scenarioStore.scenarios
};

const BottomBar = (): React.ReactElement<{}> => {
    const classes = useStyles();
    return (
        <AppBar position="fixed" color="primary" className={classes.appBar}>
            <Toolbar>
                <Grid container justify='center' className={classes.popoutButton}>
                    <Button variant="contained" color="secondary" onClick={sendScenaroisToServerAndRedirect}>Generate report</Button>
                </Grid>
            </Toolbar>
        </AppBar>
    );
};

export default BottomBar;
