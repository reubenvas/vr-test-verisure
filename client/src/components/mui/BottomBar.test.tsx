import React from 'react';
import renderer from 'react-test-renderer';
import BottomBar from './BottomBar';

describe('DarkButton component', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<BottomBar />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
