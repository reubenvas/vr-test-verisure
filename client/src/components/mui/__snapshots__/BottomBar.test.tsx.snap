// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkButton component snapshot renders 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <div
      className="MuiGrid-root makeStyles-popoutButton-2 MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Generate report
        </span>
      </button>
    </div>
  </div>
</header>
`;
